ambry-network/src/main/java/com/github/ambry/network/NettyServerRequestResponseChannel.java

```diff
@@ -59,8 +64,12 @@ public NettyServerRequestResponseChannel(NetworkConfig config, Http2ServerMetric
   /** Send a request to be handled */
   @Override
   public void sendRequest(NetworkRequest request) throws InterruptedException {
-    networkRequestQueue.offer(request);
-    http2ServerMetrics.requestEnqueueTime.update(System.currentTimeMillis() - request.getStartTimeInMs());
+    if (networkRequestQueue.offer(request)) {
```

Comment:
let's also keep a metric here to see how many requests are coming in the the channel.

Reply:
done